### PR TITLE
build(cmake/windows) Use CMake for building on Windows in CI

### PR DESCRIPTION
**Pull request by quyykk**
## Feature Details

This replaces scons+winmake with CMake when building on Windows in the CI environment.

## Testing Done

CI will test everything.

### CODE REVIEW

**Comment by quyykk** [comment]
Release is here: https://github.com/quyykk/endless-sky/releases/tag/v16windows3

I still need to test it, but it should work

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -45,45 +45,32 @@ jobs:
...
-        run: ./utils/cd_update_versions.sh
-        shell: bash
       - name: Setup sccache
         uses: ./.github/sccache
         with:
           read-only: 'true'
-      - name: Build Application
-        run: make -re -j ($(Get-CIMInstance -Class 'CIM_Processor').NumberOfLogicalProcessors) -f .winmake
+      - name: Adjust version strings
+        run: ./utils/cd_update_versions.sh
```
Comment (MCOfficer) [inline .github/workflows/cd.yaml:61]: Does this use bash by default?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -319,35 +280,37 @@ jobs:
...
-      matrix:
-        os: [windows-2022]
+      CONTINUOUS: EndlessSky-win64-continuous.zip
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
     - uses: actions/checkout@v3
-    - name: Install runtime dependencies
+    # If no code changes occurred then we can download the latest continuous.
+    - name: Download latest continuous
+      if: ${{ needs.changed.outputs.game_code != 'true' && needs.changed.outputs.cmake != 'true' }}
```
Comment (MCOfficer) [inline .github/workflows/ci.yml:296]: why not `== 'false'`?

Comment (quyykk) [reply .github/workflows/ci.yml:296]: That doesn't actually work funnily enough. Apparently it should but when I tried it the step was skipped anyways.

File: .github/workflows/cd.yaml (hunk 1 above)
Comment (quyykk) [reply .github/workflows/cd.yaml:61]: oops

**Comment by quyykk** [comment]
new release https://github.com/quyykk/endless-sky/releases/tag/v16windows4

also untested

**Pull request merged**